scipy/linalg/tests/test_lapack.py

```diff
@@ -3298,3 +3298,93 @@ def test_gges_tgsen(dtype):
 
     assert_allclose(s[0, 0] / t[0, 0], d2, rtol=0, atol=atol)
     assert_allclose(s[1, 1] / t[1, 1], d1, rtol=0, atol=atol)
+
+@pytest.mark.parametrize(
+    "a,b,c,d,e,f,r,l",
+    [(np.array( [[  4.0,  1.0,  1.0,  2.0] ,
+                 [  0.0,  3.0,  4.0,  1.0] ,
+                 [  0.0,  1.0,  3.0,  1.0] ,
+                 [  0.0,  0.0,  0.0,  6.0]]),
+      np.array( [[  1.0,  1.0,  1.0,  1.0] ,
+                 [  0.0,  3.0,  4.0,  1.0] ,
+                 [  0.0,  1.0,  3.0,  1.0] ,
+                 [  0.0,  0.0,  0.0,  4.0]]),
+      np.array( [[ -4.0,  7.0,  1.0, 12.0] ,
+                 [ -9.0,  2.0, -2.0, -2.0] ,
+                 [ -4.0,  2.0, -2.0,  8.0] ,
+                 [ -7.0,  7.0, -6.0, 19.0]]),
+      np.array( [[  2.0,  1.0,  1.0,  3.0] ,
+                 [  0.0,  1.0,  2.0,  1.0] ,
+                 [  0.0,  0.0,  1.0,  1.0] ,
+                 [  0.0,  0.0,  0.0,  2.0]]),
+      np.array( [[  1.0,  1.0,  1.0,  2.0] ,
+                 [  0.0,  1.0,  4.0,  1.0] ,
+                 [  0.0,  0.0,  1.0,  1.0] ,
+                 [  0.0,  0.0,  0.0,  1.0]]),
+      np.array( [[ -7.0,  5.0,  0.0,  7.0] ,
+                 [ -5.0,  1.0, -8.0,  0.0] ,
+                 [ -1.0,  2.0, -3.0,  5.0] ,
+                 [ -3.0,  2.0,  0.0,  5.0]]),
+      np.array( [[  1.0,  1.0,  1.0,  1.0] ,
+                 [ -1.0,  2.0, -1.0, -1.0] ,
+                 [ -1.0,  1.0,  3.0,  1.0] ,
+                 [ -1.0,  1.0, -1.0,  4.0]]),
+     np.array(  [[  4.0, -1.0,  1.0, -1.0] ,
+                 [  1.0,  3.0, -1.0,  1.0] ,
+                 [ -1.0,  1.0,  2.0, -1.0] ,
+                 [  1.0, -1.0,  1.0,  1.0]]) )])
+@pytest.mark.parametrize('dtype', REAL_DTYPES)
+def test_tgsyl_NAG(a,b,c,d,e,f,r,l,dtype):
+    atol = 1e-4
+    
+    tgsyl = get_lapack_funcs(('tgsyl'), dtype=dtype)
+    rans,lans,scale,dif,info = tgsyl(a,b,c,d,e,f)
+    
+    assert_equal(info, 0)
+    assert_almost_equal(scale,1.0,decimal=4,err_msg="SCALE must be 1.0")
```

Comment:
We try to avoid using `assert_almost_equal` and some similar functions in new code. See [Code and Documentation Style Guide - The Missing Bits](https://docs.scipy.org/doc/scipy/dev/missing-bits.html#test-functions-from-numpy-testing) for some more info.

Reply:
Addressed it in my next checkin. Also increased tolerance for single precision. Let's see if tests pass.